core/src/main/scala/cats/functor/Phantom.scala

```diff
@@ -0,0 +1,21 @@
+package cats
+package functor
+
+import simulacrum.typeclass
+
+/**
+  * Phantom-variant Functor.
+  * Values of type A must not appear in F[A].
+  * Natural result of a functor being both covariant and contravariant.
+  */
+@typeclass trait Phantom[F[_]] extends Contravariant[F] {
+  def functor: Functor[F] = new Functor[F] {
```

Comment:
I don't follow why this can't extend `Functor[F]`. If the tests have limitations, I think we should fix those (I think other test classes have multiple parents).

I know we have thought about different encodings of typeclasses, including this style. I would rather us not have one lone typeclass following a different approach.